JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/timeseries/TimeSeriesSparql.java

```diff
@@ -143,30 +360,78 @@ public boolean checkTimeUnitExists(String tsIRI) {
-     * @param timeUnit the time unit of the time series (optional)
-     */    
-    protected void initTS(String timeSeriesIRI, List<String> dataIRI, String dbURL, String timeUnit) {
-    	// Construct time series IRI
-    	Iri tsIRI;
+
+	/**
+	 * Instantiate the time series instance in the knowledge base (time unit is optional)
+	 * @param timeSeriesIRI timeseries IRI provided as string
+	 * @param dataIRI list of data IRI provided as string that should be attached to the timeseries
+	 * @param dbURL URL of the database where the timeseries data is stored provided as string
+	 * @param timeUnit the time unit of the time series (optional)
+	 * @param type type of TimeSeries data to be instantiated.
+	 *             Allowed values of type enum: Type.AVERAGE, Type.INSTANTANEOUS, Type.STEPWISECUMULATIVE, Type.CUMULATIVETOTAL
+	 * @param duration Required for Average Time Series. Numeric duration of the averaging period for Average TimeSeries of type Duration. Only positive values are allowed. (optional)
+	 * @param unit Required for Average Time Series. Temporal unit type of the averaging period for Average TimeSeries. (optional)
+	 *             Allowed values of type ChronoUnit:
+	 *             ChronoUnit.SECONDS, ChronoUnit.MINUTES, ChronoUnit.HOURS, ChronoUit.DAYS, ChronoUnit.WEEKS, ChronoUnit.MONTHS, ChronoUnit.YEARS
+	 *
+	 */
+
+	protected void initTS(String timeSeriesIRI, List<String> dataIRI, String dbURL, String timeUnit, Iri type, Duration duration, ChronoUnit unit) {
+		// Construct time series IRI
+		Iri tsIRI;
 		// Check whether timeseriesIRI follows IRI naming convention of namespace & local_name
-    	// If only local name is provided, iri() function would add filepath as default prefix
-    	// Every legal (full) IRI contains at least one ':' character to separate the scheme from the rest of the IRI
+		// If only local name is provided, iri() function would add filepath as default prefix
+		// Every legal (full) IRI contains at least one ':' character to separate the scheme from the rest of the IRI
 		if (Pattern.compile("\\w+\\S+:\\S+\\w+").matcher(timeSeriesIRI).matches()) {
 			tsIRI = iri(timeSeriesIRI);
 		} else {
 			throw new JPSRuntimeException(exceptionPrefix + "Time series IRI does not have valid IRI format");
 		}
 
-		// Check that the time series IRI is not yet in the Knowledge Graph
-		if (checkTimeSeriesExists(timeSeriesIRI)) {
-			throw new JPSRuntimeException(exceptionPrefix + "Time series " + timeSeriesIRI + " already in the Knowledge Graph");
+		ModifyQuery modify = Queries.MODIFY();
+		// set prefix declarations
+		modify.prefix(prefix_ontology, prefix_kb, prefix_time);
+
+		// define type
+		modify.insert(tsIRI.isA(type));
+
+		if(type.equals(AverageTimeSeries)){
+
+			if(duration.getNano()!=0){
+				LOGGER.warn("Nano is ignored");
+			}
+
+			//numeric Duration
+			if(duration.getSeconds() <=0.0){
+				throw new JPSRuntimeException(exceptionPrefix + "Numeric Duration must be a positive value");
+			}
```

Comment:
I think these checks are impossible to be triggered? Surely the Duration class has its own checks

Reply:
Duration allows objects like Duration.ofDays(-12) which will trigger this

Reply:
there is a method Duration.isNegative(), that will be preferable than this. I still feel that it's up to the users to store whatever they want and it shouldn't be TimeSeriesClient's responsibility to check this

Reply:
Yi Kai wanted this check to be there as he is going to be using this at the moment, that's why I added it there

Reply:
Duration.isNegative() won't cover 0 which is invalid, either duration.isNegative() and Duration.isZero() should be used together or the existing check can be left as it is.

Reply:
Then please use the isNegative method, it will be tidier